drafts/arcalinea-tx-expiry/draft.rst

```diff
@@ -0,0 +1,61 @@
+**Title:** Transaction Expiry
+
+**Author:** Jay Graber
+
+**Status:** Active
+
+**Category:** Standards
+
+**Created:** 2018-01-09
+
+Abstract
+===========
+
+This is an Standards ZIP describing a new consensus rule to expire transactions that are not mined from the mempool.
+
+Motivation
+===========
+
+Transactions that are too large or have insufficient fees are often not mined. This indeterminism is a source of confusion for users and wallets. Giving users the ability to specify the number of blocks a transaction should sit in the mempool before it expires, and otherwise setting a default, would provide certainty around how long a transaction has to confirm.
+
+Advantages include improving performance by removing transactions unlikely to be mined, and simplifying bidirectional payment channels by reducing the need to store and compress revocations for past states since transactions are allowed to expire.
+
+Specification
+===============
+
+Transactions will have a new field, nBlockExpiry, which will set the blockheight after which transactions will be removed from the mempool if they have not been mined. The last block that the transaction below could possibly be included in is 3539. After that, it will be removed from the mempool.
```

Comment:
Is there any benefit to making this `nExpiryTime`, and allowing it to be either a time or a block height, with same semantics as `nLockTime`? How well do Bitcoin users reason about `nLockTime`?

Reply:
Yeah, `nLockTime`,  is commonly used by `OP_CHECKLOCKTIMEVERIFY` scripts, think it would make sense to follow the semantics as `nExpiryTime`.

As for using both `nLockTime` and `nExpiryTime` together, I could imagine a use case where a time range for when a transaction could be committed to the chain is set. If participants are passing transactions to record state off-chain, this could allow transactions to expire if they're not published and mined within a certain time period.